src/VisualStudio/Core/Def/Implementation/TableDataSource/VisualStudioDiagnosticListTable.cs

```diff
@@ -130,5 +150,98 @@ private void OnErrorListPropertyChanged(object sender, PropertyChangedEventArgs
                 AddTableSourceIfNecessary(this.Workspace.CurrentSolution);
             }
         }
+
+        private void OnWorkspaceChanged(object sender, WorkspaceChangeEventArgs e)
+        {
+            Contract.ThrowIfFalse(_errorList is IErrorList2);
+
+            switch (e.Kind)
+            {
+                case WorkspaceChangeKind.SolutionAdded:
+                case WorkspaceChangeKind.SolutionChanged:
+                case WorkspaceChangeKind.SolutionCleared:
+                case WorkspaceChangeKind.SolutionReloaded:
+                case WorkspaceChangeKind.SolutionRemoved:
+                case WorkspaceChangeKind.ProjectAdded:
+                case WorkspaceChangeKind.ProjectChanged:
+                case WorkspaceChangeKind.ProjectRemoved:
+                    // Set error list toggle state based on current analysis state for all languages for projects in current solution.
+                    var fullAnalysisState = _optionService.GetOption(RuntimeOptions.FullSolutionAnalysis);
+                    if (fullAnalysisState)
+                    {
+                        var langauges = e.NewSolution.Projects.Select(p => p.Language).Distinct();
+                        foreach (var language in langauges)
+                        {
+                            if (!ServiceFeatureOnOffOptions.IsClosedFileDiagnosticsEnabled(_optionService, language))
+                            {
+                                fullAnalysisState = false;
+                                break;
+                            }
+                        }
+                    }
+
+                    ((IErrorList2)_errorList).AnalysisToggleState = fullAnalysisState;
+                    return;
+            }
+        }
+
+        private void OnErrorListFullSolutionAnalysisToggled(object sender, AnalysisToggleStateChangedEventArgs e)
+        {
+            var newOptions = _optionService.GetOptions()
+                .WithChangedOption(RuntimeOptions.FullSolutionAnalysis, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.CSharp, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, LanguageNames.VisualBasic, e.NewState)
+                .WithChangedOption(ServiceFeatureOnOffOptions.ClosedFileDiagnostic, TypeScriptLanguageName, e.NewState);
+            
+            _optionService.SetOptions(newOptions);
+        }
+
+        private void OnOptionChanged(object sender, OptionChangedEventArgs e)
+        {
+            Contract.ThrowIfFalse(_errorList is IErrorList2);
+
+            if (e.Option == RuntimeOptions.FullSolutionAnalysis || e.Option == ServiceFeatureOnOffOptions.ClosedFileDiagnostic)
+            {
+                var analysisDisabled = !(bool)e.Value;
+                if (analysisDisabled)
+                {
+                    ((IErrorList2)_errorList).AnalysisToggleState = false;
+                }
```

Comment:
FullSolutionAnalysis option can only be turned off by Low VM situation. we are hiding the language specific tools option from the end user (ClosedFileDiagnostic), but still ensuring that if someone turns it off via the registry key, then we change the error list button state to off.